File: front/mobile/lib/models/absence.dart
class Absence {
  final int id;
  final int employeeId;
  final int absenceTypeId;
  final String? absenceTypeName;
  final DateTime startDate;
  final DateTime endDate;
  final double daysCount;
  final String status;
  final String? reason;
  final String? rejectionReason;

  Absence({
    required this.id,
    required this.employeeId,
    required this.absenceTypeId,
    this.absenceTypeName,
    required this.startDate,
    required this.endDate,
    required this.daysCount,
    required this.status,
    this.reason,
    this.rejectionReason,
  });

  factory Absence.fromJson(Map<String, dynamic> json) {
    return Absence(
      id: json['id'] as int,
      employeeId: json['employee_id'] as int,
      absenceTypeId: json['absence_type_id'] as int,
      absenceTypeName: json['absence_type']?['name'] as String?,
      startDate: DateTime.parse(json['start_date'] as String),
      endDate: DateTime.parse(json['end_date'] as String),
      daysCount: (json['days_count'] as num).toDouble(),
      status: json['status'] as String,
      reason: json['reason'] as String?,
      rejectionReason: json['rejected_reason'] as String?,
    );
  }
}

File: api/app/Modules/Absence/Domain/Models/Absence.php
<?php

declare(strict_types=1);

namespace App\Modules\Absence\Domain\Models;

use App\Core\Auth\Domain\Models\Employee;
use App\Traits\BelongsToCompany;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * Domain model for the Absence aggregate.
 *
 * Colonnes reelles de la table `absences`
 * (migration tenant 2026_04_01_000103_create_attendance_absences_advances.php).
 *
 * @property int    $id
 * @property int|null $company_id
 * @property int    $employee_id
 * @property int    $absence_type_id
 * @property string $start_date
 * @property string $end_date
 * @property int    $days_count
 * @property string $status  (pending|approved|rejected|cancelled)
 * @property string|null $reason
 * @property string|null $proof_path
 * @property int|null $approved_by
 * @property string|null $rejected_reason
 * @mixin \Illuminate\Database\Eloquent\Builder<static>
 */
class Absence extends Model
{
    use BelongsToCompany;
    use HasFactory;

    protected $fillable = [
        'company_id',
        'employee_id',
        'absence_type_id',
        'start_date',
        'end_date',
        'days_count',
        'status',
        'reason',
        'proof_path',
        'approved_by',
        'rejected_reason',
    ];

    protected $casts = [
        'start_date' => 'date',
        'end_date'   => 'date',
    ];

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class);
    }

    public function absenceType(): BelongsTo
    {
        return $this->belongsTo(AbsenceType::class);
    }
}
